osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs

```diff
@@ -147,23 +151,48 @@ private set
 
         private bool canCurrentlyTrack => Time.Current >= slider.StartTime && Time.Current < slider.EndTime;
 
+        private OsuAction? trackingAction;
+
+        private bool cursorTrackingBall => lastScreenSpaceMousePosition.HasValue
+                                        && ReceivePositionalInputAt(lastScreenSpaceMousePosition.Value);
+
         protected override void Update()
         {
             base.Update();
 
             if (Time.Current < slider.EndTime)
             {
                 // Make sure to use the base version of ReceivePositionalInputAt so that we correctly check the position.
-                Tracking = canCurrentlyTrack
-                           && lastScreenSpaceMousePosition.HasValue
-                           && ReceivePositionalInputAt(lastScreenSpaceMousePosition.Value)
-                           && (drawableSlider?.OsuActionInputManager?.PressedActions.Any(x => x == OsuAction.LeftButton || x == OsuAction.RightButton) ?? false);
+                Tracking = cursorTrackingBall
+                        && canCurrentlyTrack
```

Comment:
Would reorder this to go `canCurrentlyTrack && cursorTrackingBall`, to save a `ToLocalSpace()` computation.